### ISSUE TITLE
spotlessCheck sometimes fails even after running spotlessApply for googleJavaFormat

### ISSUE THREAD

**Issue by jaredjstewart** [diffplug/spotless#37]
Hi,

I want to add the spotless plugin pointed to the Eclipse Google Java Style Guide xml, but right now spotlessCheck fails for one file in my project even if I run it directly after spotlessApply.  Is there any way to see what it thinks is wrong with the file?

Thank you!

**Comment by nedtwigg** [comment]
That is very surprising.  `spotlessApply` is how you see what is wrong with the file - `spotlessCheck` just checks that the file's content is the same as the end result of `spotlessApply`.

Can you copy-paste the file in question and/or the spotless part of your `build.gradle`?

**Comment by jaredjstewart** [comment]
Branch with the spotless plugin added:
https://github.com/jaredjstewart/incubator-geode/tree/spotlessPluginGoogleStyle?files=1

From build.gradle:
apply plugin: "com.diffplug.gradle.spotless"

  spotless {

    java {

      //For now I've only set up formatting, not import ordering.

      //importOrder ['java', 'javax', 'org', 'com', 'com.diffplug', ''] //
An array of package names

      //importOrderFile 'spotless.importorder' // An import ordering file,
exported from Eclipse

      eclipseFormatFile
"${rootProject.projectDir}/etc/eclipse-java-google-style.xml"

    }

  }

**Comment by jaredjstewart** [comment]
If it helps, here is the formatter: 
https://github.com/jaredjstewart/incubator-geode/blob/spotlessPluginGoogleStyle/etc/eclipse-java-google-style.xml

Here is the class not being formatted correctly:
https://github.com/jaredjstewart/incubator-geode/blob/spotlessPluginGoogleStyle/geode-core/src/test/java/org/apache/geode/cache30/Bug38741DUnitTest.java

And here is the build.gradle: 
https://github.com/jaredjstewart/incubator-geode/blob/spotlessPluginGoogleStyle/build.gradle

**Comment by nedtwigg** [comment]
I just duplicated your issue.  You're using version `1.3.3`, and upgrading to the latest `2.2.0` fixes the problem.

I'm not sure what went wrong, and I'm curious to find out, but there's been a lot of improvements since `1.3.0`, so I'm not soo curious as to dig it up.  But thanks for the report, and if you have any other issues in the future post another issue :)

Btw, if you want the "canonical" google style, and you don't care about the eclipse code formatter, you might want to take a look at [this section of the readme](https://github.com/diffplug/spotless#applying-to-java-source-google-java-format).

**Comment by jaredjstewart** [comment]
Fantastic, thank you so much!

**Comment by jaredjstewart** [comment]
By the way, when I bumped to the latest version it worked with the Google Java Style eclipse XML.  But when I tried switching to googleJavaStyle() I had the same issue (with different files) where spotlessCheck would fail directly after spotlessApply.

**Comment by nedtwigg** [comment]
Thanks very much for letting me know, I'll dig in!

**Comment by nedtwigg** [comment]
After a little digging, I've traced this to a bug in the `google-java-format`.  I'm adding a feature to Spotless to handle misbehaving rules...

**Comment by jaredjstewart** [comment]
Cool!  By the way, the Apache Geode community started using Spotless to enforce code formatting as of this afternoon :)

**Comment by nedtwigg** [comment]
Just wanted to let you know that I used your case of `spotlessCheck` fails right after calling `spotlessApply` to build a [padded cell mode](https://github.com/diffplug/spotless/blob/master/PADDEDCELL.md) which resolves exactly this kind of issue, and also generates a bug report for the rule in question.

I've filed [a bug with google-java-format](https://github.com/google/google-java-format/issues/89), so hopefully they'll be able to fix it in a future release.

Is it alright if I list apache-geode as an example project in the README?
